docs/docsite/rst/dev_guide/overview_architecture.rst

```diff
@@ -10,26 +10,34 @@ It uses no agents and no additional custom security infrastructure, so it's easy
 
 In this section, we'll give you a really quick overview of how Ansible works so you can see how the pieces fit together.
 
+.. contents::
+   :local:
+
 Modules
 =======
 
-Ansible works by connecting to your nodes and pushing out small programs, called "Ansible Modules" to them. These programs are written to be resource models of the desired state of the system. Ansible then executes these modules (over SSH by default), and removes them when finished.
+Ansible works by connecting to your nodes and pushing out small programs, called "Ansible Modules" to them. These programs are written to be resource models of the desired state of the system. Ansible then executes these modules (over SSH by default), and removes them when finished. Your library of modules can reside on any machine, and there are no servers, daemons, or databases required.
```

Comment:
>  These programs are written to be resource models of the desired state of the system.

I'm not sure what "resource models" means.... I think this sentence is trying to say that most modules are written to take parameters specifying the desired state of the system rather than to take parameters specifying an action to perform on the system.  This is just convention (for instance, the shell and command modules break that convention) but is a goal with the modules that we've written.

Reply:
+1 to the rewording.